addons/hr_timesheet/migrations/10.0.1.0/pre-migration.py

```diff
@@ -29,7 +29,32 @@ def create_and_populate_department(cr):
     ''')
 
 
+def create_and_populate_children(cr):
+    openupgrade.logged_query(cr, '''
+    ALTER TABLE project_task ADD COLUMN parent_id INT;
+
+    -- Since the parent_id is new there's no point in computing the
+    -- children_hours, they will be 0; however there's a single method
+    -- computing children_hours, effective_hours, remaining_hours,
+    -- total_hours, total_hours_spent, and delay_hours.  However, only
+    -- children_hours and total_hours_spent need recomputation.
+
+    -- These are new columns, the others existed before but created in another
+    --  addon.
+
+    ALTER TABLE project_task ADD COLUMN children_hours DOUBLE PRECISION;
```

Comment:
As stated by @hbrunn in https://github.com/OCA/openupgradelib/pull/93#issue-162200382, these fields won't have XML-ID, so we should change approach of pre-creating columns having an openupgradelib method that makes the rest of the work (create the column, create the field and assign an XML-ID).

Reply:
I didn't realize that when I made the PR.   I guess I should have used the `add_ir_model_fields` function.  Am I right?

Reply:
Now that I read better is not `add_ir_model_fields`.

Reply:
@mvaled I have created the method in this PR: https://github.com/OCA/openupgradelib/pull/100

Reply:
Hi @pedrobaeza   

Thanks for the new method. I made a rebase on 10.0 and squash all the commits into one.